memory/index.go

```diff
@@ -284,3 +298,27 @@ func (idx *Index) columnIndexes(schema sql.Schema) []int {
 	}
 	return indexes
 }
+
+// contentHashedFields returns a slice of field indexes in this secondary index that should be hashed, instead
+// of directly storing their content. This is only applicable to unique secondary indexes.
+func (idx *Index) contentHashedFields() (contentHashedFields []uint) {
+	if !idx.Unique {
+		return nil
+	}
+
+	for i, expr := range idx.Exprs {
+		if !types.IsTextBlob(expr.Type()) {
```

Comment:
is `StringType.Length()` the same info?

Reply:
I don't think so... length alone is enough to identify this case. For example, `TINYTEXT` is a valid `TEXT` type, but its length is 255, whereas MySQL allows a `varchar` type to have a length up to 65,535, so length alone couldn't distinguish the TEXT/BLOB type there.